boudicca.events/eventcollectors/src/main/kotlin/events/boudicca/eventcollector/collectors/CafeTraxlmayrCollector.kt

```diff
@@ -74,36 +74,57 @@ class CafeTraxlmayrCollector : TwoStepEventCollector<Element>("cafetraxlmayr") {
     }
 
     private fun parseLesungen(event: Element, builder: StructuredEvent.StructuredEventBuilder): List<StructuredEvent> {
-        return event.select(".modal-body div")
-            .eachText()
-            .mapNotNull {
-                if (!it.contains("Uhr")) {
-                    null
+        val contentBlocks = event.select(".modal-body > *")
+        var startDate = OffsetDateTime.MIN
+        var name = ""
+        var description = ""
+        var pictureSrc = ""
+        val events = ArrayList<StructuredEvent>();
+
+        for (block in contentBlocks) {
+            val text = block.text()
+            if (block.select("img").isNotEmpty()){
+                pictureSrc = block.select("img").attr("src")
+            } else if (text.contains("Uhr")) {
+                startDate = parseDateForLesung(text)
+            } else if (text.contains("„")) {
+                name = text
+            } else {
+                description += text
+            }
+            if (startDate != OffsetDateTime.MIN && name.isNotEmpty() && description.isNotEmpty() && pictureSrc.isNotEmpty()) {
+                val pictureUrl = if (pictureSrc.isNotBlank()) {
+                    UrlUtils.parse(baseUrl + pictureSrc)
                 } else {
-                    val dateTimeSplit = it.split(": ")
-                    val startDate = parseDateForLesung(dateTimeSplit[0])
-                    val name = dateTimeSplit[1]
-                    builder
-                        .copy()
-                        .withName(name)
-                        .withStartDate(startDate)
-                        .build()
+                    null
                 }
+                events.add(builder
+                    .withName(name)
```

Comment:
if you resuse a eventbuilder please make a copy before doing so with .copy()! otherwise data from one event could leak to another event, when the second event does not overwrite it

Reply:
thanks for the info, i added a copy() to the builder!